docs/java/release-notes.mdx

```diff
@@ -27,21 +27,78 @@ We highly recommend regularly updating to the latest SDK version. It will help y
 - update client libraries giving access to latest SAP services on SAP Cloud Platform and S4/HANA
 - protect yourself from bugs and breaking changes in the future
 
+## 3.29.1 - September 24, 2020
+
+[Maven Central](https://search.maven.org/search?q=g:com.sap.cloud.sdk*%20AND%20v:3.29.1)
```

Comment:
[Release blog](https://blogs.sap.com/?p=1189923)

We're still missing the Javadocs link